JavaScript/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -145,12 +176,35 @@ module Microsoft.ApplicationInsights {
 
             sessionStateEnvelope.time = Util.toISOStringForIE8(new Date(timestamp));
 
-            tc._track(sessionStateEnvelope); 
+            tc._track(sessionStateEnvelope);
+        }
+
+        private _applyCustomProperties(envelope: Microsoft.Telemetry.Envelope) {
+            if (this.properties && envelope && envelope.data && (<any>envelope.data).baseData) {
+                var telemetryItem = (<any>envelope.data).baseData;
+                telemetryItem.properties = telemetryItem.properties || {};
+                Object.getOwnPropertyNames(this.properties).forEach((val, index, array) => {
+                    if (telemetryItem.properties[val] === undefined) {
+                        telemetryItem.properties[val] = this.properties[val];
+                    }
+                });
+            }
+        }
+
+        private _applyCustomMeasurements(envelope: Microsoft.Telemetry.Envelope) {
+            if (this.measurements && envelope && envelope.data && (<any>envelope.data).baseData) {
```

Comment:
_applyCustomProperties and _applyCustomMeasurements are identical functions with the exception of source and target object you're populating. Perhaps create a single function that takes in these two objects as input parameters and call this function from both _applyCustomProperties and _applyCustomMeasurements